src/components/miscellaneous/ShareButton.tsx

```diff
@@ -0,0 +1,40 @@
+// Material UI Components
+import Button from '@material-ui/core/Button';
+
+import useShare from 'use-share';
+
+export type ShareProps = {
+  title?: string;
+  type: 'event' | 'news' | 'jobpost' | 'pages';
+  id: number | string;
+};
+
+const ShareButton = ({ type, id, title }: ShareProps) => {
+  let typeToURL = '';
+
+  switch (type) {
+    case 'event':
+      typeToURL = 'a';
+      break;
+    case 'news':
+      typeToURL = 'n';
+      break;
+    case 'jobpost':
+      typeToURL = 'k';
+      break;
+    case 'pages':
+      typeToURL = 'om';
+      break;
+  }
```

Comment:
Du kan endre dette til å bruke `useMemo` slik at du slipper `let` og heller bruker `const`:
```suggestion
  const urlType = useMemo(() => {
    switch (type) {
      case 'event':
        return 'a';
      case 'news':
        return 'n';
      case 'jobpost':
        return 'k';
      case 'pages':
        return 'om';
    }
  }, [type]);
```

`useMemo` kan importeres fra `react`. Den brukes ved å legge inn en funksjon som returnerer en verdi og en array som er `dependencies` som får funksjonen til å kjøres på nytt når en dependency endres.